controllers/autoscaling.go

```diff
@@ -18,30 +19,61 @@ func (r *HorizontalRunnerAutoscalerReconciler) determineDesiredReplicas(rd v1alp
 
 	var repos [][]string
 
-	repoID := rd.Spec.Template.Spec.Repository
-	if repoID == "" {
-		orgName := rd.Spec.Template.Spec.Organization
-		if orgName == "" {
-			return nil, fmt.Errorf("asserting runner deployment spec to detect bug: spec.template.organization should not be empty on this code path")
-		}
+	orgName := rd.Spec.Template.Spec.Organization
+	if orgName == "" {
+		return nil, fmt.Errorf("asserting runner deployment spec to detect bug: spec.template.organization should not be empty on this code path")
+	}
 
-		metrics := hra.Spec.Metrics
+	metrics := hra.Spec.Metrics
+	if len(metrics) == 0 {
+		return nil, fmt.Errorf("validating autoscaling metrics: one or more metrics is required")
+	} else if tpe := metrics[0].Type; tpe != v1alpha1.AutoscalingMetricTypeTotalNumberOfQueuedAndInProgressWorkflowRuns {
+		return nil, fmt.Errorf("validting autoscaling metrics: unsupported metric type %q: only supported value is %s", tpe, v1alpha1.AutoscalingMetricTypeTotalNumberOfQueuedAndInProgressWorkflowRuns)
+	}
```

Comment:
I would actually add another metrics type here instead of overwriting the previous implementation. What do you think about what I've written in https://github.com/summerwind/actions-runner-controller/pull/223/files#diff-2fb5cc41bbf098ed839a501baea824efe104844213ce3969a63523439a132600R31

Reply:
That way https://github.com/summerwind/actions-runner-controller/pull/213/files#diff-2fb5cc41bbf098ed839a501baea824efe104844213ce3969a63523439a132600R41 wont break deployments on github enterprise

Reply:
Awesome work you did on your PR!

So when I look at the new `calculateReplicasByPercentageRunnersBusy` method you introduce, you are essentially fetching all the runners and subsequently iterate over that list by checking which one is busy. Next, you have a couple of conditionals that instructs you to either scale up or down.

I like it because this also allows us to get completely rid of checking queues on an individual repository level. What do you reckon to do next here? I haven't given your PR a very thorough review yet, but with yours merged I guess we can close this one? Or do we want to refactor the changes a bit by cherry-picking your changes and the logic for the `calculateReplicasByQueuedAndInProgressWorkflowRuns` method/metric and keep the kind of ugly `if len(metrics[0].RepositoryNames) < 1 && r.GitHubClient.GithubEnterprise` conditional?

Reply:
I actually think your hpa scheme is also very much needed. I recommend you create another hpa scheme and add it into this if else block and pull out your changes into a separate func. Something like `v1aplha1.AutoscalingMetricTypeNumQueuedAndInProgressForEnabledRepositories` and then leave the original scheme as is.

I don't mind keeping these as separate PRs and if yours goes in first, merge your changes into mine afterwards. What are your thoughts?